lib/EndpointUtil.js

```diff
@@ -0,0 +1,106 @@
+	 *
+	 * */
+	static extractPlatformIdAndOptions(platformIdOrRegion, options) {
+		let _platformId;
+		let _options = options;
+
+		if (options && typeof options !== 'object' || Array.isArray(options)  ){ // falsy values are fine too
+			throw new TypeError(`options needs to be an object or falsy if specified, received: ${typeof options} | ${options}`);
+		}
+
+		if (!platformIdOrRegion){
+			// making sure falsy values do not prevent the default to be set later on
+			_platformId = undefined;
+		} else if (typeof platformIdOrRegion === 'object'){
+			// only options given
+			_options = platformIdOrRegion;
+		} else if (typeof platformIdOrRegion === 'string'){
+			_platformId = LeagueUtil.getPlatformIdFromPlatformIdOrRegion(platformIdOrRegion);
+		}
+
+		return {_platformId, _options};
+	}
+
+	static buildQueryStringFromOptions(options) {
+		let paramsArray = Object.keys(options).map(optionKey => {
+			const option = options[optionKey];
+
+			if (Array.isArray(option)) {
+				return option.map(v => {
+					return `${optionKey}=${v}`;
+				}).join('&');
+			} else {
+				return `${optionKey}=${option}`;
+			}
+		});
+		return paramsArray.join('&');
+	}
+
+	static throwIfParamIsArray(param, paramName) {
+		if (param instanceof Array) {
+			throw new ParameterError(`${paramName} can not be an Array. Received: ${param}`);
+		}
+	}
+
+	static throwIfNotNumerical(param, paramName) {
+		if (isNaN(parseInt(param))) {
+			throw new TypeError(`${paramName} has to be a number or numerical string. Received: ${param}`);
+		}
+	}
+
+	static throwIfInvalidPlatformId(platformId) {
+		if (!LeagueUtil.validatePlatformId(platformId)) {
+			throw new ParameterError(`${platformId} is not a valid platformId`);
+		}
+	}
+
+	static throwIfNotBoolean(param, paramName) {
+		const isBooleanString = typeof param === 'string' && (param === 'true' || param === 'false');
+		const isBoolean = param === true || param === false;
+		if (!(isBoolean || isBooleanString)) {
+			throw new TypeError(`${paramName} has to be a boolean or booleanlike string (true,false,'true','false'). Received: ${param}`);
+		}
+	}
+	static throwIfNameIsInvalid(name) {
+		if (!LeagueUtil.validateSummonerNameInputLength(name)) {
+			throw new Error(`Summoner name "${name}" is too long (16 characters max, found ${name.length})`);
+		}
+		const invalidCharactersMatch = LeagueUtil.validateSummonerNameInputCharacters(name);
+		if (invalidCharactersMatch) {
+			throw new Error(`Summoner name "${name}" contains invalid characters: ${invalidCharactersMatch.join(' | ')}`);
+		}
+	}
+
+	static throwIfRankedQueueConfigIdInvalid(rankedQueueConfigId) {
+		const validRankedQueueConfigIds = ['RANKED_SOLO_5x5', 'RANKED_FLEX_TT', 'RANKED_FLEX_SR'];
+		if (validRankedQueueConfigIds.indexOf(rankedQueueConfigId) === -1) {
+			throw new ParameterError(`rankedQueueConfigId has to be one of ${validRankedQueueConfigIds}. Received: ${rankedQueueConfigId}`);
+		}
+	}
+
+	static throwIfNotObject(param, paramName) {
```

Comment:
I understand.
I think it would be better to just wrap them into Promises within the Endpoint methods.
Otherwise those Methods would become pseudo asynchron, which is not neccessary.

When wrapping them into the Promise returned by the endpoints that Promise will get rejected in case they throw.